core/src/main/scala/io/projectglow/vcf/BigVCFDatasource.scala

```diff
@@ -58,6 +62,40 @@ object BigVCFDatasource extends HlsUsageLogging {
         .contains(VCFFileWriter.INFER_HEADER))) {
       throw new SparkException("Cannot infer header for empty VCF.")
     }
+
+    val extraInferredOptions = if (options.getOrElse(
+        VCFFileWriter.VCF_HEADER_KEY,
+        VCFFileWriter.INFER_HEADER) == VCFFileWriter.INFER_HEADER) {
+      // If we have to infer the header (explicitly or implicitly), determine all of the samples in the DataFrame.
+      // The samples that will be written consist of the distinct set of sample IDs identified across the DataFrame,
+      // in addition to missing sample IDs. The number of missing sample IDs will be equal to the maximum number
+      // found in any given row.
+      import data.sparkSession.implicits._
+      val distinctSampleLists =
+        data
+          .select("genotypes.sampleId")
+          .as[Array[String]]
+          .distinct()
```

Comment:
Does `distinct` work correctly on array columns?

Reply:
It may be easier to reason about / more efficient to do
```
df.selectExpr("collect_set(explode(genotypes.sampleId))")`
```

Reply:
`distinct` does work correctly on array columns. If we explode the sample IDs, I don't believe we'll be able to know how many sample IDs we're missing.

Reply:
Also once we explode, there will be a single sample ID per row so there's not much use in calling `collect_set` on the `String`; I assume you meant `sampleIdDf.distinct()`.